src/engine/GuidanceLine.cs

```diff
@@ -85,11 +102,21 @@ public override void _Process(double delta)
 
         dirty = false;
         mesh.ClearSurfaces();
-        mesh.SurfaceBegin(Mesh.PrimitiveType.Lines);
+        mesh.SurfaceBegin(Mesh.PrimitiveType.Triangles);
 
         mesh.SurfaceSetColor(colour);
-        mesh.SurfaceAddVertex(LineStart);
+        
+        // To form quad, we want it in 'orgin + vector' form, not 'start + end' form
+        // Be sure to flatten the Y-axis of the vector, so it's all on a 2D plane
+        Vector3 lineVector = lineEnd - lineStart;
+        lineVector[1] = 0.0f;
+        float lineVectorLength = (float)Math.Sqrt(lineVector[0]*lineVector[0] + lineVector[2]*lineVector[2]);
```

Comment:
Why not use the length (or was it Distance) property available in the vector class?
Also same thing a few lines below, there's a `.Normalized()` method that returns a normalized variant.

Reply:
Oh and does this handle the case where the line ends up being exactly 0 length? Some player is probably going to run into that at some point in the future and then make a bug report if the line behaves strange in that situation.

Reply:
I tried, but then it was throwing me a bunch of errors as if the Vector3 didn't have those methods. I suppose it would be more proper for me to figure out *why* it wasn't working, so the code would be cleaner. I can work to revise that.

Reply:
Yeah, that sounds very strange. As-is this code doesn't fit in with the rest of the codebase because there's 0 of this manual vector math anywhere else, it's all done using the vector class methods to make it clear what is being done.

Reply:
Okay; I'll work on a new version that both uses the built-in vector math and tests if the length is zero first. Thank you.